renpy/common/00action_other.rpy

```diff
@@ -734,6 +734,28 @@ init -1500 python:
             renpy.clear_capture_focus(self.name)
             renpy.restart_interaction()
 
+    @renpy.pure
+    class ToggleFocus(Action, DictEquality):
+        """
+        :doc: focus_action
+
+        If the focus exist, clears it, otherwise captures it.
+        """
+
+        def __init__(self, name="default"):
+            self.name = name
+
+        def __call__(self):
+            name = self.name
+
+            if renpy.get_focus_rect(name) is not None:
+                renpy.clear_capture_focus(name)
+
+            else:
+                renpy.capture_focus(name)
+
+            renpy.restart_interaction()
```

Comment:
Based purely on the existing actions, does this only need to happen when clearing, rather than always?

Reply:
Ah, I was reading the `GetFocusRect` action. Nevermind. /o\